crates/shim/src/asynchronous/mod.rs

```diff
@@ -388,13 +386,73 @@ fn signal_server_started() {
     }
 }
 
+#[cfg(unix)]
+fn signal_stream(kind: i32) -> std::io::Result<BoxStream<'static, i32>> {
+    use tokio::signal::unix::{signal, SignalKind};
+    let kind = SignalKind::from_raw(kind);
+    signal(kind).map(|mut sig| {
+        use std::task::{ready, Poll};
+
+        use futures::stream::poll_fn;
+        poll_fn(move |cx| -> Poll<Option<i32>> {
+            ready!(sig.poll_recv(cx));
+            Poll::Ready(Some(kind.as_raw_value()))
```

Comment:
I'm not sure I understand this part. Why do we poll manually here?

Reply:
An alternative would be to use `tokio-stream`'s wrappers for Signals and CtrlC: https://docs.rs/tokio-stream/latest/tokio_stream/wrappers/struct.SignalStream.html

I don't think it's worth adding a new dependency when `poll_fn` works fine, but it would be a clearer without having to add a comment.